File: javascript/prob_001_two_number_sum/solution_01.js
function two_number_sum(array, target_sum) {
  map = {};

  for (var i = 0; i < array.length; i++) {
    item = array[i];
    other = target_sum - item;
    if (map[other]) {
      return [item, other];
    } else {
      map[item] = true;
    }
  }

  return [];
}

console.log(two_number_sum([1,3,4,5,2,24,11], 4));

File: python/easy/prob_001_two_number_sum/solution_01.py
# O(n) time
# O(n) space
def two_number_sum(array, target_sum):
  map_items = {}
  for item in array:
    other = target_sum - item
    if other in map_items:
      return [item, other]
    else:
      map_items[item] = True
  return []

print(two_number_sum([1, 3, 4, 222, 5], 6))
